packages/ibm-products/src/components/ConditionBuilder/ConditionBuilderItem/ConditionBuilderItemOption/ConditionBuilderItemOption.js

```diff
@@ -54,6 +55,18 @@ export const ConditionBuilderItemOption = ({
     // eslint-disable-next-line react-hooks/exhaustive-deps
   }, []);
 
+  useEffect(() => {
+    //this will focus the first input field in the popover
+
+    if (contentRef.current) {
+      const firstFocusableElement =
+        contentRef.current?.querySelector('input, button,li');
+
+      firstFocusableElement?.focus();
+    }
+    // eslint-disable-next-line react-hooks/exhaustive-deps
```

Comment:
we shouldn't be disabling this kind of warning. is there a reason specifically for doing this?

Reply:
hey,
I am expecting this useEffect to run only once at the time of rendering (on load). If I pass any dependency , the behaviour will is not expected.
_lint error: React Hook useEffect has missing dependencies: 'allOptions', 'conditionState', and 'getOptions'. Either include them or remove the dependency array_